chore: NEXT-40616 - undeprecate onlyAvailable flag (#6885)
shopware/platform: UPGRADE-6.8.md

- `\Shopware\Core\Content\Sitemap\Event\SitemapRouteCacheKeyEvent`
- `\Shopware\Core\Content\Sitemap\Event\SitemapRouteCacheTagsEvent`

## `filterByActiveRules` in Payment- and ShippingMethodCollection removed

> The `filterByActiveRules` methods in `Shopware\Core\Checkout\Payment\PaymentMethodCollection` and `Shopware\Core\Checkout\Shipping\ShippingMethodCollection` are deprecated.

Reviewer: ```suggestion
The `filterByActiveRules` methods in `Shopware\Core\Checkout\Payment\PaymentMethodCollection` and `Shopware\Core\Checkout\Shipping\ShippingMethodCollection` were removed.
```

At this point those methods were already removed. The deprecation announcement has to be written in the 6.7 upgrade guide.

Use the new `Shopware\Core\Framework\Rule\RuleIdMatcher` instead.

> It allows filtering of RuleIdAware objects in either Arrays or Collections.

Reviewer: ```suggestion
It allows filtering of `RuleIdAware` objects in either arrays or collections.
```
